pkg/procs/info_test.go

```diff
@@ -0,0 +1,115 @@
+package procs_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"reflect"
+	"testing"
+
+	"github.com/openshift-kni/debug-tools/pkg/procs"
+)
+
+var nullLog = log.New(ioutil.Discard, "", 0)
+
+func TestEmpty(t *testing.T) {
+	dir, err := ioutil.TempDir("", "test")
+	if err != nil {
+		t.Fatalf("creating temp dir %v", err)
+	}
+	defer os.RemoveAll(dir) // clean up
+
+	ph := procs.New(nullLog, dir)
+	pidInfos, err := ph.ListAll()
+	if err != nil {
+		t.Errorf("ListAll(%s) failed: %v", dir, err)
+	}
+	if len(pidInfos) > 0 {
+		t.Errorf("found unexpected entries: %v", pidInfos)
+	}
+}
+
+func TestSingleProcSingleThread(t *testing.T) {
+	dir, err := ioutil.TempDir("", "test")
+	if err != nil {
+		t.Fatalf("creating temp dir %v", err)
+	}
+	defer os.RemoveAll(dir) // clean up
+
+	if err := makeFakeTree(dir, map[int]fakeEntry{
+		1: fakeEntry{
+			attrs: fakeAttrs{
+				"cmdline": "/usr/lib/systemd/systemd\x00--switched-root\x00--system\x00--deserialize",
+			},
+			tasks: map[int]fakeAttrs{
+				1: fakeAttrs{
+					"status": "Name:	systemd\nPid:	1\nCpus_allowed_list:	0-3\n",
```

Comment:
how dependent are we on whitespaces/tabs here ?  can it ruin the test if mishandled ?

Reply:
while the business logic should not actually care, we need to use whatever the real file in `/proc` is using.